docs/src/sharing.rst

```diff
@@ -0,0 +1,183 @@
+   limitations under the License.
+
+.. _sharing:
+
+#############################################
+Adding External Information to Your |cdk| App
+#############################################
+
+.. See https://github.com/awslabs/aws-cdk/issues/603 (includes work from the following PR)
+       https://github.com/awslabs/aws-cdk/pull/183
+
+There are a number of ways you can get information into your app from outside your app.
+
+- Using a command-line context variable
+- Using an environment variable
+- Using an SSM Parameter Store variable
+- Using a value from another stack
+- Using a |CFN| template value
+
+Each of these techniques is described in the following sections.
+
+.. _sharing_from_command_line:
+
+Getting a Value from a Command-Line Context Variable
+====================================================
+
+To pass the value of a command-line context value to your app,
+run a |toolkit| command using the **--context** (**-c**) option,
+as shown in the following example.
+
+.. code-block:: sh
+
+    cdk synth -c bucket_name=mygroovybucket
+
+To get the value of a command-line context value in your app,
+use code like the following,
+which gets the value of the context variable **bucket_name**.
+
+.. code-block:: ts
+
+    const bucket_name string = this.getContext("bucket_name");
+
+.. _sharing_from_env_vars:
+
+Getting a Value from an Environment Variable
+============================================
+
+To get the value of an environment variable,
+use code like the following,
+which gets the value of the environment variable **MYBUCKET**.
+
+.. code-block:: ts
+
+    const bucket_name = process.env.MYBUCKET;
+
+.. _sharing_from_ssm:
+
+Getting a Value from an SSM Store Variable
+==========================================
+
+To get the value of an SSM parameter store variable,
+use code like the following,
+which uses the value of the SSM variable **my-awesome-value**.
+
+.. code-block:: ts
+		
+    const myvalue = new SSMParameterProvider(this).getString("my-awesome-value");
+
+See the :doc:`context` topic for information about the :py:class:`SSMParameterProvider <@aws-cdk/cdk.SSMParameterProvider>`.
+
+.. _sharing_between_stacks:
+
+Sharing Information Between Stacks
+==================================
+
+You can share information between two stacks in the same app
+by using the **export** method on one stack and the **import** method on the other stack.
+
+First add a property to the class that defines the stack you are exporting from.
+The following example shows a stack with the property **myBucketRefProps**.
+
+.. code-block:: ts
+
+    class HelloCdkStack extends cdk.Stack {
+        public readonly myBucketRefProps: s3.BucketRefProps;
+
+	constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
+            super(parent, name, props);
+
+Next create a bucket and export it's *BucketRefProps* to the **myBucketRefProps** property.
```

Comment:
I feel all the intermediate text here adds more complication than that it enlightens.

Can't we just show the complete exporting stack, and the complete importing stack in a complete code snippet?

Basically go:

This is how you export:

***CODE SNIPPET***

Import like this:

***CODE SNIPPET***

Connect like this:

***CODE SNIPPET***